clients/java/src/main/java/io/zeebe/client/api/command/GatewayConnectionException.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.zeebe.client.api.command;
+
+public class GatewayConnectionException extends ClientException {
+
+  private static final long serialVersionUID = -5195750018378722095L;
+
+  public GatewayConnectionException(Throwable cause) {
+    super(String.format("Cannot connect to the gateway. Because: %s", cause.getMessage()));
```

Comment:
What would be great is if we knew the expected gateway endpoint here, so we could write something like:

`Expected to connect to ADDRESS, but the gateway is currently unavailable. Cause: %s`

I guess that would need some redesign (since this future impl isn't meant to know about the client right now, I suppose), so that's out.

How about:

```suggestion
    super(String.format("Unable to connect to the gateway, as it is currently unavailable. Depending on the cause, this may be a transient error. Cause: %s", cause.getMessage()));
```